btsensor/src/bthome.rs

```diff
@@ -0,0 +1,637 @@
+            Self::GasDetected => "gas detected",
+            Self::HeatAbnormal => "abnormal heat",
+            Self::LightDetected => "light detected",
+            Self::Unlocked => "unlocked",
+            Self::Wet => "wet",
+            Self::MotionDetected => "motion detected",
+            Self::Moving => "moving",
+            Self::OccupancyDetected => "occupancy detected",
+            Self::Plugged => "plugged",
+            Self::Present => "present",
+            Self::Problem => "problem",
+            Self::Running => "running",
+            Self::Safe => "safe",
+            Self::SmokeDetected => "smoke detected",
+            Self::Sound => "sound",
+            Self::Tamper => "tampered",
+            Self::VibrationDetected => "vibration detected",
+            Self::WindowOpen => "window open",
+            Self::ButtonEvent => "button event",
+            Self::DimmerEvent => "dimmer event",
+        }
+    }
+
+    pub fn unit(self) -> &'static str {
+        match self {
+            Self::PacketId
+            | Self::Count
+            | Self::Timestamp
+            | Self::GenericBoolean
+            | Self::PowerOn
+            | Self::Open
+            | Self::BatteryLow
+            | Self::BatteryCharging
+            | Self::CarbonMonoxideDetected
+            | Self::Cold
+            | Self::Connected
+            | Self::DoorOpen
+            | Self::GarageDoorOpen
+            | Self::GasDetected
+            | Self::HeatAbnormal
+            | Self::LightDetected
+            | Self::Unlocked
+            | Self::Wet
+            | Self::MotionDetected
+            | Self::Moving
+            | Self::OccupancyDetected
+            | Self::Plugged
+            | Self::Present
+            | Self::Problem
+            | Self::Running
+            | Self::Safe
+            | Self::SmokeDetected
+            | Self::Sound
+            | Self::Tamper
+            | Self::VibrationDetected
+            | Self::WindowOpen
+            | Self::ButtonEvent
+            | Self::DimmerEvent => "",
+            Self::Battery
+            | Self::Humidity
+            | Self::HumidityShort
+            | Self::Moisture
+            | Self::MoistureShort => "%",
+            Self::Temperature => "°C",
+            Self::Pressure => "hPa",
+            Self::Illuminance => "lux",
+            Self::MassKg => "kg",
+            Self::MassLb => "lb",
+            Self::Dewpoint => "°C",
+            Self::Energy => "kWh",
+            Self::Power => "W",
+            Self::Voltage => "V",
+            Self::Pm2_5 | Self::Pm10 | Self::Tvoc => "ug/m3",
+            Self::Co2 => "ppm",
+            Self::Acceleration => "m/s²",
+            Self::Gyroscope => "°/s",
+        }
+    }
+
+    /// The number of spaces to the left to move the decimal point.
+    ///
+    /// In other words, the value stored should be divided by 10 to the power of this number to get
+    /// the actual value.
+    fn decimal_point(self) -> i32 {
```

Comment:
I would probably call this `denominator` or something, and make it return an integer or float (1.0 or 100.0 or 1000.0)

Reply:
Done.